modules/data.remote/R/Lantrendr_AGB_prep.R

```diff
@@ -0,0 +1,132 @@
+  if(length(new_site_info$site_id) != 0){
+    if(is.null(buffer) | as.logical(skip_buffer)){
+      #extracting AGB data
+      med_agb_data <- PEcAn.data.remote::extract.LandTrendr.AGB(new_site_info, "median", fun = "mean", 
+                                                                AGB_input_dir, product_dates=lubridate::year(start_date):lubridate::year(end_date))[[1]] %>% dplyr::select(-2) %>%
+        `colnames<-`(c("site_id", paste0(time_points, "_AbvGrndWood")))
+      sdev_agb_data <- PEcAn.data.remote::extract.LandTrendr.AGB(new_site_info, "stdv", fun = "mean", 
+                                                                 AGB_input_dir, product_dates=lubridate::year(start_date):lubridate::year(end_date))[[1]]%>% dplyr::select(-c(1:2)) %>%
+        `colnames<-`(c(paste0(time_points, "_SD")))
+      #Handle data
+      AGB_Output <- cbind(med_agb_data, sdev_agb_data)
+    }else{#buffer is not empty
+      #extracting AGB data
+      med <- PEcAn.data.remote::extract.LandTrendr.AGB(new_site_info, "median", buffer = buffer, fun = "mean", 
+                                                                AGB_input_dir, product_dates=lubridate::year(start_date):lubridate::year(end_date))
+      sdev <- PEcAn.data.remote::extract.LandTrendr.AGB(new_site_info, "stdv", buffer = buffer, fun = "mean", 
+                                                                 AGB_input_dir, product_dates=lubridate::year(start_date):lubridate::year(end_date))
+      sdev_agb_data <- med_agb_data <- c()
+      for (i in seq_along(new_site_info$site_id)) {
+        temp_var <- rowSums(sdev[[i]])
+        min_var_Ind <- which.min(temp_var)
+        
+        sdev_agb_data <- rbind(sdev_agb_data, sdev[[i]][min_var_Ind,])
+        med_agb_data <- rbind(med_agb_data, med[[i]][min_var_Ind,])
+      }
+      colnames(med_agb_data) <- paste0(time_points, "_AbvGrndWood")
+      colnames(sdev_agb_data) <- paste0(time_points, "_SD")
+      #Handle data
+      AGB_Output <- cbind(med_agb_data, sdev_agb_data) %>% as.data.frame
+      AGB_Output$site_id <- site_info$site_id
+    }
+    
+    #prepare CSV from AGB_Output
+    Current_CSV <- matrix(NA, 0, 6) %>% `colnames<-`(c("date", "site_id", "lat", "lon", "agb", "sd"))
+    for (id in AGB_Output$site_id) {
+      site_AGB <- AGB_Output[which(AGB_Output$site_id==id),]
+      for (i in seq_along(time_points)) {
+        date <- lubridate::year(time_points[i])
+        site_id <- id
+        lon <- new_site_info$lon[which(new_site_info$site_id==id)]
+        lat <- new_site_info$lat[which(new_site_info$site_id==id)]
+        agb <- site_AGB[paste0(time_points[i], "_AbvGrndWood")] %>% purrr::set_names("agb")
+        sd <- site_AGB[paste0(time_points[i], "_SD")] %>% purrr::set_names("sd")
+        Current_CSV <- rbind(Current_CSV, tibble::tibble(date, site_id, lat, lon, agb, sd))#in date, id, lat, lon, agb, sd
+      }
+    }
+
+    #Compare with existing CSV file. (We name the CSV file as LAI.csv)
```

Comment:
The CSV naming convention is necessary because it tells people where this variable comes from and where it will go (saved as a CSV file). Another naming convention, like AGB_output, differs from the CSV naming that the AGB_output will be passed to the assembler function and will not be saved like the CSV file.